aptos-move/framework/move-stdlib/sources/configs/features.move

```diff
@@ -350,13 +362,33 @@ module std::features {
         if (!exists<Features>(@std)) {
             move_to<Features>(framework, Features { features: vector[] })
         };
-        let features = &mut borrow_global_mut<Features>(@std).features;
-        vector::for_each_ref(&enable, |feature| {
-            set(features, *feature, true);
-        });
-        vector::for_each_ref(&disable, |feature| {
-            set(features, *feature, false);
-        });
+        apply_diff(borrow_global_mut<Features>(@std), enable, disable);
+    }
+
+    /// Function to enable and disable features. Can only be called by a signer of @std.
+    public fun change_feature_flags_for_next_epoch(framework: &signer, enable: vector<u64>, disable: vector<u64>)
+    acquires Features {
+        assert!(signer::address_of(framework) == @std, error::permission_denied(EFRAMEWORK_SIGNER_NEEDED));
+        let features = if (config_for_next_epoch::does_exist<Features>()) {
+            config_for_next_epoch::extract<Features>(framework)
+        } else if (exists<Features>(@std)) {
+            *borrow_global<Features>(@std)
+        } else {
+            Features { features: vector[] }
+        };
+        apply_diff(&mut features, enable, disable);
+        config_for_next_epoch::upsert(framework, features);
+    }
+
+    /// Apply all the pending feature flag changes. Should only be used at the end of a reconfiguration with DKG.
+    ///
+    /// While the scope is public, it can only be usd in system transactions like `block_prologue` and governance proposals,
+    /// who have permission to set the flag that's checked in `extract()`.
+    public fun on_new_epoch(account: &signer) acquires Features {
+        if (config_for_next_epoch::does_exist<Features>()) {
+            let features = config_for_next_epoch::extract<Features>(account);
+            *borrow_global_mut<Features>(@std) = features;
```

Comment:
This looks really weird. Does this actually work? You should destruct the features returned from config_for_next_epoch::extract and replace the inner vector value with it.

Reply:
yeah that's related to the drop ability. what's bad about replacing the whole resource?